Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs

```diff
@@ -0,0 +1,155 @@
+using Content.Shared.Database;
+using Content.Shared.Hands.Components;
+using Content.Shared.Item;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Robust.Shared.Physics;
+
+namespace Content.Shared.Hands.EntitySystems;
+
+public abstract partial class SharedHandsSystem : EntitySystem
+{
+    /// <summary>
+    ///     Tries to pick up an entity to a specific hand.
+    /// </summary>
+    public bool TryPickup(EntityUid uid, EntityUid entity, string? handName = null, bool checkActionBlocker = true, bool animateUser = false, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        var hand = handsComp.ActiveHand;
+        if (handName != null && !handsComp.Hands.TryGetValue(handName, out hand))
+            return false;
+
+        if (hand == null)
+            return false;
+
+        return TryPickup(uid, entity, hand, checkActionBlocker, animateUser, handsComp, item);
+    }
+
+    public bool TryPickupAnyHand(EntityUid uid, EntityUid entity, bool checkActionBlocker = true, bool animateUser = false, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        if (!TryGetEmptyHand(uid, out var hand, handsComp))
+            return false;
+
+        return TryPickup(uid, entity, hand, checkActionBlocker, animateUser, handsComp, item);
+    }
+
+    public bool TryPickup(EntityUid uid, EntityUid entity, Hand hand, bool checkActionBlocker = true, bool animateUser = false, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        if (!Resolve(entity, ref item, false))
+            return false;
+
+        if (!CanPickup(uid, entity, hand, checkActionBlocker, handsComp, item))
+            return false;
+
+        // animation
+        var xform = Transform(uid);
+        var coordinateEntity = xform.ParentUid.IsValid() ? xform.ParentUid : uid;
+        var initialPosition = EntityCoordinates.FromMap(coordinateEntity, Transform(entity).MapPosition);
```

Comment:
cant you pass entman into these or something

Reply:
yes you can, will fix.